File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double max_avg=INT_MIN, sum=0;
        int n=nums.size();
        for(int i=0; i<k; i++){
            sum+=nums[i];
        }
        max_avg=sum;
        for(int i=0; i<n-k; i++){
            sum+=nums[i+k]-nums[i];
            max_avg = max(max_avg, sum);
        }
        return max_avg/k;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {    
    let currSum = 0;
    
    for (let i = 0; i < k; i++) {
        currSum += nums[i];
    }
    
    let maxSum = currSum;
    
    for (let i = k; i < nums.length; i++) {
        currSum += nums[i] - nums[i - k];
        maxSum = Math.max(currSum, maxSum);
    }
    
    return maxSum / k;
};